### ISSUE TITLE
Very Hard Immigration Bug

### ISSUE THREAD

**Issue by jfkonecn** [bvschaik/julius#262]
I know the goal of this project to keep the game logic the way it was so that people have the option to play Caesar 3 with only UI updates, but there's a bug which I really hurts the user experience.

On very hard when the population is between 200 and 300. Your citizens will randomly become angry with you and start leaving your city. This results in you having to do silly things to quickly rush 100 people into your city just to get past the bug.

Steps to reproduce
1. Set game to very hard difficultly
2. Have 200 people in your city
See between 11:56 - 15:04
https://youtu.be/DIfvCtI3OXc?t=716

The fix is a one liner
https://github.com/jfkonecn/julius/commit/787080c186b33871c0ebd42c998290f915e238f8

Basically your base housing happiness is 40 when you play on very hard.
Before you hit a population of 300 only base housing happiness is accounted for.
So no matter what you do, your overall happiness is 40. If you happiness is below 50 then people start leaving your city.

This, however, only happens after a population of 200 because the game gives you a boost to prevent this immigration problem from happening to you in the first place. If I had to guess, I think the creators meant to have the boost until 300, but since only crazy people play on very hard no one caught the bug before release.

**Comment by crudelios** [comment]
I have to say I want this one fixed too.

In fact I fixed it in a branch of my own a year ago, but never submitted a PR.

**Comment by 411752230** [comment]
Agree. A hardcore difficulty, which should have been fixed long times ago ,could be soften to softcore difficulty. No matter what difficulty it is, it should be well-acceptable and playable among players.

**Comment by ArtySh0ck** [comment]
Agree, but i think, to save Very Hard diff, that emigration glitch must be swapped with another hard thing, you know, like much more angry gods or people if wages low / taxes high. It makes sense to do people are not pretty happy when u pay Wages = Rome's wages, yours must be set as least for one more (Rome's wages is 32, so you have to pay 33). And taxes, if you tax more than 6% - people are more angry. Why its important? Im no master of c3, but even me have found c3 a quite easy at Very Hard, and that emigration bug was only reason to feel some perspiration while play, sure only for start, but its something. Sure it has to be fixed, but we need another Harderer of game, PLEASE. Thanks.

**Comment by KillahNeo** [comment]
I'm not sure Bianca will listen to this type of tip, she doesn't want to fix any bugs on the game which may cause old savegames breaking.

But if she did, we'd have a decent Caesar III version, the one we deserve and never got

**Comment by jfkonecn** [comment]
This, to me, is the same thing as fixing the UI bug which places a road next to a gatehouse after you delete the road next to it then rotate your view. There are ways around it and it doesn’t impact the game overall. It just really hurts the user experience and it is something that was not done intentionally by the developers.
